server/bin/pbench-audit-server.py

```diff
@@ -579,49 +585,32 @@ def main():
         ihier = IncomingHierarchy("incoming", config.INCOMING, config)
         cihier = ControllerHierarchy("incoming", config.INCOMING, config)
         verify_controllers(ihier, cihier)
-        cnt = 0
-        c_check = cihier.check_controller_in_controller()
-        i_check = ihier.check_controller_in_IRHierarchy()
-        if c_check or i_check:
-            cihier.header(f, "start")
-            if c_check:
-                cnt = cihier.dump(f)
-            if i_check:
-                cnt += ihier.dump(f)
-            cihier.header(f, "end")
+        cnt = check_and_dump(f, cihier, ihier)
         if cnt > 0:
             ret += 1
 
         rhier = ResultsHierarchy("results", config.RESULTS, config)
+        """
+        crhier being a ControllerHierarchy object needs to be created
+        again as this object consists of tarball/file details of results
+        hierarchy which are completely different from the other hierarchy.
+        """
```

Comment:
I don't think this set is really an issue, since the parameters are clearly different from the earlier ControllerHierarchy (INCOMING vs RESULTS). It's more of an issue between these and the objects further down, where we appear to be constructing new objects with identical parameters ("results", config.RESULTS, config) and no obvious reason that the meaning of those (e.g., the config dict) would have changed between the two calls.

While the comment suggests we're now including user hierarchy, isn't not clear how that is established within the constructors (at least, not without a lot of digging; I guess I need to go back and look more closely). Also, if these ARE different, then this suggests another coupling between the classes and the mainline code which is a maintenance and documentation issue we should resolve!

Reply:
We can do one thing. Mainly verifylist of CONTROLLER hierarchy is the one which leads to verify_incoming and verify_results. We can return verifylist to the main and call them(verify_incoming and verify_results) with verifylist as a function argument.

Reply:
it will also lose hierarchy_dispatcher

Reply:
and also lose verify_hierarchy function